### PR TITLE
Login networking

### PR DESCRIPTION
**Pull request by guvanch-amangeldiyev**
login with validation

### CODE REVIEW

File: packages/data/pubspec.lock [hunk 1]
```diff
@@ -0,0 +1,196 @@
+# Generated by pub
```
Comment (ilya-safyan-idf) [inline packages/data/pubspec.lock]: gitignore

File: packages/data/lib/dio/dio_builder.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    connectTimeout: timeout,
+    receiveTimeout: timeout,
+    sendTimeout: timeout,
+    headers: headersMap,
+    contentType: 'application/json',
+  );
+
+  final dio = Dio(options);
+  dio.interceptors.add(LogInterceptor(requestBody: true, responseBody: true));
+  // dio.interceptors.add(CancelInterceptor());
```
Comment (ilya-safyan-idf) [inline packages/data/lib/dio/dio_builder.dart]: delete comments

File: packages/domain/lib/usecase/usecase.dart [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// abstract class UseCase<Output> {
+//   Output call();
```
Comment (ilya-safyan-idf) [inline packages/domain/lib/usecase/usecase.dart]: delete comments

File: packages/domain/pubspec.lock [hunk 4]
```diff
@@ -0,0 +1,175 @@
+# Generated by pub
```
Comment (ilya-safyan-idf) [inline packages/domain/pubspec.lock]: gitignore

File: packages/presentation/lib/screen/home/home_page.dart [hunk 5]
```diff
@@ -0,0 +1,41 @@
+// import 'package:flutter/material.dart';
+// import 'package:flutter/widgets.dart';
```
Comment (ilya-safyan-idf) [inline packages/presentation/lib/screen/home/home_page.dart]: delete comments

File: packages/presentation/pubspec.lock [hunk 6]
```diff
@@ -0,0 +1,238 @@
+# Generated by pub
```
Comment (ilya-safyan-idf) [inline packages/presentation/pubspec.lock]: gitignore

File: packages/data/lib/core/api_key.dart [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  static const String baseUrl = "http://jenkins-mobile.moneyman.ru";
+  static const String pathUrl = "/api/json?pretty=true";
+  static const String password = "1Aa_Bb";
+  static const String login = "flutter_lab_test";
+  static const String pathToken = "/crumbIssuer/api/json";
+  static Map<String, dynamic> authHeader = {
+    "Authorization": "Basic ${base64.encode(utf8.encode("$login:$password"))}"
+  };
+  static String headerString =
+      "Basic ${base64.encode(utf8.encode("$login:$password"))}";
```
Comment (ilya-safyan-idf) [inline packages/data/lib/core/api_key.dart]: formatting - add commas before the last brackets

File: packages/data/lib/dio/dio_builder.dart [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    baseUrl: baseUrl,
+    connectTimeout: timeout,
+    receiveTimeout: timeout,
+    sendTimeout: timeout,
+    headers: headersMap,
+    contentType: 'application/json',
+  );
+
+  final dio = Dio(options);
+  dio.interceptors.add(LogInterceptor(requestBody: true, responseBody: true));
```
Comment (ilya-safyan-idf) [inline packages/data/lib/dio/dio_builder.dart]: formatting - add commas before the last brackets

File: packages/data/pubspec.lock (hunk 1 above)
Comment (guvanch-amangeldiyev) [reply packages/data/pubspec.lock]: added

File: packages/data/lib/dio/dio_builder.dart (hunk 2 above)
Comment (guvanch-amangeldiyev) [reply packages/data/lib/dio/dio_builder.dart]: deleted

File: packages/domain/lib/usecase/usecase.dart (hunk 3 above)
Comment (guvanch-amangeldiyev) [reply packages/domain/lib/usecase/usecase.dart]: deleted

File: packages/presentation/lib/screen/home/home_page.dart (hunk 5 above)
Comment (guvanch-amangeldiyev) [reply packages/presentation/lib/screen/home/home_page.dart]: ok

**Pull request merged**